pkg/acsengine/params_k8s.go

```diff
@@ -244,7 +254,13 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 				CloudProviderRateLimitBucket: properties.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket,
 			})
 			addValue(parametersMap, "kubeClusterCidr", properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet)
-			addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.OrchestratorProfile.KubernetesConfig.KubeletConfig["--non-masquerade-cidr"])
+			if properties.MasterProfile != nil && properties.MasterProfile.IsCustomVNET() {
+				addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.MasterProfile.VnetCidr)
+			} else if properties.HostedMasterProfile != nil {
+				// TODO add AKS kubernetesNonMasqueradeCidr value here
+			} else {
+				addValue(parametersMap, "kubernetesNonMasqueradeCidr", DefaultVNETCIDR)
```

Comment:
This also introduce slightly different behavior since in the kubenet case, the original code will use subnet for masq, but here it will use vnet.